ledger-core/virtual/integration/method_test.go

```diff
@@ -1728,3 +1732,191 @@ func TestVirtual_CallContractTwoTimes(t *testing.T) {
+				require.Equal(t, []byte("finish B.Bar"), result)
+			},
+			&execution.Update{
+				Type:   execution.Done,
+				Result: requestresult.New([]byte("finish A.Foo"), objectAGlobal),
+			},
+		)
+
+		runnerMock.AddExecutionClassify("Foo", flags, nil)
+	}
+
+	// add checks to typedChecker
+	{
+		typedChecker.VStateReport.Set(func(report *payload.VStateReport) bool {
+			assert.True(t, report.DelegationSpec.IsZero())
+
+			switch report.Object {
+			case objectAGlobal:
+				require.Equal(t, int32(0), report.OrderedPendingCount)
+				require.Equal(t, int32(1), report.UnorderedPendingCount)
+			case objectBGlobal:
+				require.Equal(t, int32(0), report.OrderedPendingCount)
+				require.Equal(t, int32(0), report.UnorderedPendingCount)
+			default:
+				t.Fatal("unknown object")
+			}
+
+			return false
+		})
+		typedChecker.VDelegatedCallRequest.Set(func(request *payload.VDelegatedCallRequest) bool {
+			assert.Equal(t, objectAGlobal, request.Callee)
+			assert.Equal(t, outgoingRefGlobal, request.CallOutgoing)
+			assert.True(t, request.DelegationSpec.IsZero())
+
+			result := payload.VDelegatedCallResponse{
+				Callee:       request.Callee,
+				CallIncoming: request.CallIncoming,
+				ResponseDelegationSpec: payload.CallDelegationToken{
+					TokenTypeAndFlags: payload.DelegationTokenTypeCall,
+					PulseNumber:       server.GetPulse().PulseNumber,
+					Callee:            request.Callee,
+					Outgoing:          request.CallOutgoing,
+					DelegateTo:        server.JetCoordinatorMock.Me(),
+				},
+			}
+			server.SendPayload(ctx, &result)
+
+			return false
+		})
+		typedChecker.VCallRequest.Set(func(request *payload.VCallRequest) bool {
+			assert.Equal(t, objectBGlobal, request.Callee)
+			assert.Equal(t, outgoingCallReason, request.CallReason)
+			assert.Equal(t, []byte("call B.Bar"), request.Arguments)
+			assert.Equal(t, endPulse, request.CallOutgoing.Pulse())
+
+			assert.False(t, request.DelegationSpec.IsZero())
+			assert.Equal(t, objectAGlobal, request.DelegationSpec.Callee)
+			assert.Equal(t, endPulse, request.DelegationSpec.PulseNumber)
+			assert.Equal(t, outgoingRefGlobal, request.DelegationSpec.Outgoing)
+
+			result := payload.VCallResult{
+				CallType:        request.CallType,
+				CallFlags:       request.CallFlags,
+				Caller:          request.Caller,
+				Callee:          request.Callee,
+				CallOutgoing:    request.CallOutgoing,
+				ReturnArguments: []byte("finish B.Bar"),
+			}
+			server.SendPayload(ctx, &result)
+
+			return false
+		})
+		typedChecker.VCallResult.Set(func(res *payload.VCallResult) bool {
+			assert.Equal(t, objectAGlobal, res.Callee)
+			assert.Equal(t, []byte("finish A.Foo"), res.ReturnArguments)
+			assert.Equal(t, outgoingRef, res.CallOutgoing)
+
+			assert.False(t, res.DelegationSpec.IsZero())
+			assert.Equal(t, objectAGlobal, res.DelegationSpec.Callee)
+			assert.Equal(t, endPulse, res.DelegationSpec.PulseNumber)
+			assert.Equal(t, outgoingRefGlobal, res.DelegationSpec.Outgoing)
+
```

Comment:
there should be token check here and in VDelegatedRequestFinished

Reply:
what do you mean? check other fields of `res.DelegationSpec`?

Reply:
my bad, misread the code. Why not create token outside and compare it in one go?

Reply:
I don't want to check all fields